src/main/java/software/amazon/cloudwatchlogs/emf/model/MetricDirective.java

```diff
@@ -86,7 +86,7 @@ void putMetric(String key, double value, Unit unit) {
      */
     void setDimensions(List<DimensionSet> dimensionSets) {
         shouldUseDefaultDimension = false;
-        dimensions = dimensionSets;
+        dimensions = new ArrayList<>(dimensionSets);
```

Comment:
This will cause an array copy and an unnecessary allocation. If ArrayList is required here, then we should probably accept it in the method signature rather than allocate a new one. However, that is a breaking change. So, I'm fine with this fix for now.

https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/19fb8f93c59dfd791f62d41f332db9e306bc1422/src/java.base/share/classes/java/util/ArrayList.java#L178-L189

Reply:
I think we should fix the issue at the MetricsContext.java class, as the issue was introduced by [the usage of Arrays.asList](https://github.com/awslabs/aws-embedded-metrics-java/blob/master/src/main/java/software/amazon/cloudwatchlogs/emf/model/MetricsContext.java#L179). We can change to use `Lists.newArrayList` in the Guava library. We need add guava to our dependencies to use this library.

Reply:
> This will cause an array copy and an unnecessary allocation. If ArrayList is required here, then we should probably accept it in the method signature rather than allocate a new one. However, that is a breaking change. So, I'm fine with this fix for now.

ArrayList isn't required. List is required -- at least a List implementation that has all of its methods implemented. Plus, would you want to directly accept a List and then hold a reference to it? Then the person who gave you the list can modify it while you're holding on to it.

> I think we should fix the issue at the MetricsContext.java class, as the issue was introduced by the usage of Arrays.asList. We can change to use Lists.newArrayList in the Guava library. We need add guava to our dependencies to use this library.

I considered making the change to MetricsContext. To me it's more brittle. Changing the `setDimension` implementation fixes the problem once and for all. Changing 1 call site fixes it just for that case. For example I have a test that does `metricDirective.setDimensions(Collections.singletonList(DimensionSet.of("Version", "1")));` that would still break if I just change the call site.

Reply:
Thanks for the prompt feedback. Let me know what y'all think and I'll send a new revision.

Reply:
That sounds reasonable to me. :+1

Reply:
Thanks. I had to update the commit, and then make a new remote branch since the hash has changed. I don't know enough about pull requests to know how to reference the new branch in this pull request. So I made a new one https://github.com/awslabs/aws-embedded-metrics-java/pull/63